### ISSUE TITLE
Lint/RequireParentheses false positive when logical-op in `[]`

### ISSUE THREAD

**Issue by Tietew** [rubocop/rubocop#11028]
## Expected behavior

No offences should be detected in following code:

``` ruby
foo[baz && zot ? 1 : 2]
foo[baz || zot ? 1 : 2]
foo[baz ? 1 : 2]
```

## Actual behavior

```
test.rb:1:1: W: Lint/RequireParentheses: Use parentheses in the method call to avoid confusion about precedence.
foo[baz && zot ? 1 : 2]
^^^^^^^^^^^^^^
test.rb:2:1: W: Lint/RequireParentheses: Use parentheses in the method call to avoid confusion about precedence.
foo[baz || zot ? 1 : 2]
^^^^^^^^^^^^^^
```

Additionaly the locations of markers are confusing.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2) [x86_64-linux]
```
